test/parallel_api/iterator/input_data_sweep_host_iter.pass.cpp

```diff
@@ -0,0 +1,82 @@
+// -*- C++ -*-
+//===-- input_data_sweep_host_iter.pass.cpp -------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/utils.h"
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+
+#include "input_data_sweep.h"
+
+#include "support/utils_invoke.h"
+
+//This test is written without indirection from invoke_on_all_hetero_policies to make clear exactly which types
+// are being tested, and to limit the number of types to be within reason.
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+template <typename T, int __recurse, typename Policy>
+void
+test(Policy&& policy, T trash, size_t n, const std::string& type_text)
+{
+    if (TestUtils::has_types_support<T>(policy.queue().get_device()))
+    {
+
+        TestUtils::usm_data_transfer<sycl::usm::alloc::shared, T> copy_out(policy.queue(), n);
+        oneapi::dpl::counting_iterator<int> counting(0);
+        // host iterator
+        std::vector<T> host_iter(n);
```

Comment:
In addition to having a `std::vector` host iterator test, would it be useful you think to have a `std::vector` with USM allocator test as well?

Reply:
As far as coverage of input data processing code, I believe this ends up being equivalent to USM pointers in practice. 
I will confirm this, and if so, I'd prefer not to add it to keep the number of permutations down.  If there is some different code invoked as a result of such a case, we should include it.

Reply:
I've added tests to cover this, and in the process uncovered that `std::vectors` `usm_allocator` were being handled inefficiently by our input data processing.  This should be rectified with  #1438.